lib/api/core/plugins/pluginContext.js

```diff
@@ -177,6 +182,37 @@ function execute (request, callback) {
 }
 
 /**
+ * Returns a currified version of pluginsManager.trigger. The pluginName param
+ * is injected in the returned function as it is pre-pended to the custom event
+ * name. This is done to avoid colliding with the kuzzle native events.
+ *
+ * @param  {String} pluginName The name of the plugin calling trigger.
+ * @return {Function}          A trigger function that makes some checks on the
+ *                             event name and prepends the plugin name to the
+ *                             event name.
+ */
+function curryTrigger (pluginName) {
+  /**
+   * @this   {Kuzzle}
+   * @param  {String} eventName The name of the custom event to trigger.
+   * @param  {Object} payload   The payload of the event.
+   */
+  return function trigger (eventName, payload) {
+    const kuzzle = this;
+
+    if (eventName.indexOf(':') !== -1) {
+      kuzzle.pluginsManager.trigger(
+        'log:error',
+        new PluginImplementationError(`Custom event invalid name (${eventName}). Colons are not allowed in custom events.`)
```

Comment:
because events can contain _one and only one colon_. And the generated event name already contains a colon (`plugin-${pluginName}:${eventName}`)